platform/viewer/src/lib/DICOMSR/srUtils.js

```diff
@@ -0,0 +1,71 @@
+import OHIF from '@ohif/core';
+
+const { studyMetadataManager } = OHIF.utils;
+
+const supportedSopClassUIDs = [
+  '1.2.840.10008.5.1.4.1.1.88.22',
+  '1.2.840.10008.5.1.4.1.1.11.1',
+];
+
+const getAllDisplaySets = () => {
+  const allStudies = studyMetadataManager.all();
+  let allDisplaySets = [];
+
+  allStudies.forEach(study => {
+    if (study.getDisplaySets) {
+      allDisplaySets = allDisplaySets.concat(study.getDisplaySets());
+    }
+  });
+
+  return allDisplaySets;
+};
+
+const getInstanceMetadata = (displaySets, sopInstanceUid) => {
+  let instance;
+
+  // find the correct instance
+  displaySets.some(displaySet => {
+    // Search the display set to find the instance metadata for
+    return (
+      displaySet.images &&
+      displaySet.images.find(instanceMetadata => {
+        if (instanceMetadata._sopInstanceUID === sopInstanceUid) {
+          instance = instanceMetadata;
+
+          return true;
+        }
+      })
+    );
+  });
+
+  return instance;
+};
+
+const getLatestSRSeries = () => {
```

Comment:
`findMostRecentStructuredReport`?
Can we pass in `studyMetadataManager` instead of importing it?
Or if we know that this needs a studies array, can this thing just take a studies array? No need to pass it something so heavy when we need so little.

Reply:
Code is updated with suggested changes.